lib/mixins/properties-mixin.html

```diff
@@ -0,0 +1,208 @@
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../utils/boot.html">
+<link rel="import" href="../utils/mixin.html">
+<link rel="import" href="../mixins/properties-changed.html">
+
+<script>
+(function() {
+  'use strict';
+
+  /**
+   * Mixin that provides a minimal starting point to using the PropertiesChanged
+   * mixin by providing a mechanism to declare properties in a static
+   * getter (e.g. static get properties() { return { foo: String } }). Changes
+   * are reported via the `_propertiesChanged` method.
+   *
+   * This mixin provides no specific support for rendering. Users are expected
+   * to create a ShadowRoot and put content into it and update it in whatever
+   * way makes sense. This can be done in reaction to properties changing by
+   * implementing `_propertiesChanged`.
+   *
+   * @mixinFunction
+   * @polymer
+   * @appliesMixin Polymer.PropertiesChanged
+   * @memberof Polymer
+   * @summary Mixin that provides a minimal starting point for using
+   * the PropertiesChanged mixin by providing a declarative `properties` object.
+   */
+   Polymer.PropertiesMixin = Polymer.dedupingMixin(superClass => {
+
+    /**
+     * @constructor
+     * @extends {superClass}
+     * @implements {Polymer_PropertiesChanged}
+     */
+    const base = Polymer.PropertiesChanged(superClass);
+
+    /**
+     * Mixes `moreProps` into `props` but upgrades shorthand type
+     * syntax to { type: Type}.
+     *
+     * @param {Object} props Properties to normalize
+     * @return {Object} Copy of input `props` with normalized properties that
+     * are in the form {type: Type}
+     * @private
+     */
+    function normalizeProperties(props) {
+      const output = {};
+      for (let p in props) {
+        const o = props[p];
+        output[p] = (typeof o === 'function') ? {type: o} : o;
+      }
+      return output;
+    }
+
+    /**
+     * Returns the super class constructor for the given class, if it is an
+     * instance of the PropertiesMixin.
+     *
+     * @param {PropertiesMixinConstructor} constructor PropertiesMixin constructor
+     * @return {PropertiesMixinConstructor} Super class constructor
+     */
+    function superPropertiesClass(constructor) {
+      const superCtor = Object.getPrototypeOf(constructor);
+      if (superCtor.prototype instanceof PropertiesMixin) {
+        return superCtor;
+      }
+    }
+
+    /**
+     * Returns a memoized version of the `properties` object for the
+     * given class. Properties not in object format are converted to at
+     * least {type}.
+     *
+     * @param {PropertiesMixinConstructor} constructor PropertiesMixin constructor
+     * @return {Object} Memoized properties object
+     */
+    function ownProperties(constructor) {
```

Comment:
This can also move out of the mixin function. There maybe other cases in the other mixins.

Reply:
Makes sense; need to check with @azakus, it might need to stay in the closure so jscompiler has access to `PropertiesMixinConstructor` assuming it's supposed to be `PropertiesMixin` (since that symbol no longer exists) but not entirely sure if it needs to be some outside type)... Will confirum with Dan on Mon.

Reply:
oh yeah... I forgot the comment might need to be in scope :(